metals/src/main/scala/scala/meta/internal/metals/TargetData.scala

```diff
@@ -50,6 +51,8 @@ final class TargetData {
   // if workspace contains symlinks, original source items are kept here and source items dealiased
   val originalSourceItems: util.Set[AbsolutePath] =
     ConcurrentHashSet.empty[AbsolutePath]
+  val sourceItemFiles: util.Set[AbsolutePath] =
```

Comment:
not too happy with creating another random set here. I was thinking of keeping some more information about source items, maybe in its original BSP form as `Map[AbsolutePath, SourceItem]` that could also be used for distinguishing generated dirs



This version has the advantage of being very straight forward and since the vast majority of source items are directories, the set is very small

what do you think?

Reply:
> I was thinking of keeping some more information about source items, maybe in its original BSP form as Map[AbsolutePath, SourceItem] that could also be used for distinguishing generated dirs

I was considering adding those additional data (like `SourceItem`) to `buildTargetSources` :)
https://github.com/scalameta/metals/blob/9fe87ba4bd2abaf41a352e4cb7146df45535b78d/metals/src/main/scala/scala/meta/internal/metals/TargetData.scala#L41-L42

While creating another random set here wouldn't be a big problem, can we stay `path.isScalaOrJava` here https://github.com/scalameta/metals/pull/3758/files#diff-314e34520d936536cd836f653612edc262cf2afb106ae5072025978256a1fa8cL83 ? As the argument of the `collect` method is from `buildTarget.sourceItems`, they are obviously a source item of the build target.

Or can we use `buildTargetSources`? (By somehow giving a pair of `AbsolutePath` and `BuildTargetIdentifier` to the `collect` method).

Reply:
The reason for going from `path.isScalaOrJava` to getting info from BSP is that some tests were failing, because of a `*.sbt` source file file that failed that check. However, it still needs to be watched as a file.